### PR TITLE
docs: revamp intro & add new "why we built" section

### PR DESCRIPTION
**Pull request by leeederek**
## Description:
Revamp the introduction and delete what we had in the "what is Kurtosis" page

## Is this change user facing?
YES

### CODE REVIEW

File: docs/docs/explanations/why-we-built-kurtosis.md [hunk 1]
```diff
@@ -1,17 +1,21 @@
 ---
```
Comment (galenmarchetti) [inline docs/docs/explanations/why-we-built-kurtosis.md:1]: @leeederek @mieubrisse  I would opt to just remove this page entirely. When we're ready to post a roadmap communicating product vision publicly, we can add the content back into the docs in a way thats directly relevant to users (they will be able to internalize where we're going, give feedback). 

Given new philosophy of being **specific** and **current** with our product, I'd say this page no longer fits into the docs (could be good for a blog post)

Comment (leeederek) [reply docs/docs/explanations/why-we-built-kurtosis.md:1]: Caught up with Galen - we concluded that this content is valuable to leave as is to help convey the vision/purpose of Kurtosis

**Comment by leeederek** [comment]
Closing this, will open a new PR

**Comment by leeederek** [comment]
captured in #409

**Pull request closed without merging**